frontend/src/metabase/visualizations/visualizations/Heading/Heading.unit.spec.tsx

```diff
@@ -206,6 +206,26 @@ describe("Text", () => {
           screen.getByDisplayValue("Variable: {{variable}}"),
         ).toBeInTheDocument();
       });
+
+      it("should call onUpdateVisualizationSettings on blur", () => {
+        const mockOnUpdateVisualizationSettings = jest.fn();
+        const options = {
+          settings: getSettingsWithText("text"),
+          isEditing: true,
+          onUpdateVisualizationSettings: mockOnUpdateVisualizationSettings,
+        };
+        setup(options);
+
+        userEvent.click(
+          screen.getByTestId("editing-dashboard-heading-preview"),
+        );
+        userEvent.type(screen.getByRole("textbox"), "foo");
+        userEvent.tab();
+
+        expect(mockOnUpdateVisualizationSettings).toHaveBeenCalledWith({
+          text: "textfoo",
+        });
+      });
```

Comment:
If `onUpdateVisualizationSettings` can have major performance impact, it could be beneficial to assert its only called once here

Reply:
f4c7e42778bcc7f5665f54cbabe34083f040b0d9